util/s3/s3Helper.go

```diff
@@ -59,37 +64,121 @@ func loadS3Keys(keyFile string) (keys s3Keys, err error) {
-	// the environment so they're not publicly-available.
-	// Download keys are loaded from updatekey.json
+func getS3UploadBucket() (bucketName string) {
+	bucketName, found := os.LookupEnv("S3_UPLOAD_BUCKET")
+	if !found {
+		bucketName = s3DefaultUploadBucket
+	}
+	return
+}
 
-	ex, err := os.Executable()
-	if err != nil {
+func getS3ReleaseBucket() (bucketName string) {
+	bucketName, found := os.LookupEnv("S3_RELEASE_BUCKET")
+	if !found {
+		bucketName = s3DefaultReleaseBucket
+	}
+	return
+}
+
+func getS3Region() (region string) {
+	region, found := os.LookupEnv("S3_REGION")
+	if !found {
+		region = s3DefaultRegion
+	}
+	return
+}
+
+func getAWSCredentials() (awsID string, awsKey string) {
+	awsID, _ = os.LookupEnv("AWS_ACCESS_KEY_ID")
+	awsKey, _ = os.LookupEnv("AWS_SECRET_ACCESS_KEY")
+	return
+}
+
+func validateS3Params(action string, awsID string, awsKey string, awsBucket string) (err error) {
+	if awsID == "" || awsKey == "" {
+		err = fmt.Errorf("unable to %s. Credentials must be specified in AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY", action)
+		return
+	}
+	if awsBucket == "" {
+		err = fmt.Errorf("unable to %s, bucket name is empty", action)
 		return
 	}
-	baseDir := filepath.Dir(ex)
-	keyFile := filepath.Join(baseDir, "updatekey.json")
+	return
+}
 
-	keys, err := loadS3Keys(keyFile)
+// UploadFileStream sends file as stream to s3
+func (helper *Helper) UploadFileStream(filename string, reader io.Reader) error {
+	uploader := s3manager.NewUploader(helper.session)
+	_, err := uploader.Upload(&s3manager.UploadInput{
+		Bucket: aws.String(helper.bucket),
+		Key:    aws.String(filepath.Base(filename)),
+		Body:   reader,
+	})
 	if err != nil {
-		err = fmt.Errorf("unable to access remote store - error loading keys: %v", err)
+		return err
+	}
+	return nil
+}
+
+// MakeS3SessionForDownloadWithBucket download with bucket name provided
+func MakeS3SessionForDownloadWithBucket(bucket string) (helper Helper, err error) {
+	if bucket == "" {
+		return MakePublicS3SessionForDownload()
+	}
+	return makePrivateS3SessionForDownload(bucket)
+}
+
+// MakeS3SessionForUploadWithBucket upload with bucket name provided
+func MakeS3SessionForUploadWithBucket(bucket string) (helper Helper, err error) {
+	if bucket == "" {
+		return makeTravisS3SessionForUpload()
+	}
+	return makeLocalS3SessionForUpload(bucket)
+}
+
+func makeLocalS3SessionForUpload(awsBucket string) (helper Helper, err error) {
+	awsID, awsKey := getAWSCredentials()
+	err = validateS3Params("upload", awsID, awsKey, awsBucket)
+	if err != nil {
+		return
+	}
+	creds := credentials.NewStaticCredentials(awsID, awsKey, "")
+	return makeS3Session(creds, awsBucket)
+}
+
+func makeTravisS3SessionForUpload() (helper Helper, err error) {
+	return makeLocalS3SessionForUpload(getS3ReleaseBucket())
+}
+
+// MakePublicS3SessionForDownload returns an s3.Helper for the default algorand S3 bucket - for downloading
+func MakePublicS3SessionForDownload() (helper Helper, err error) {
```

Comment:
s3Helper should have no specializations - it should be generic, taking credentials, region, and bucket names.  We can export constants for the default credentials and/or buckets but that logic should be external.

Reply:
I would prefer doing that in a followup PR, consolidating the 3x `s3Helper.go` files already has a lot of moving parts without refactoring them at the same time.

Reply:
Removed the conditional logic from the S3Helper.